extensions/cache/runtime/src/main/java/io/quarkus/cache/CacheName.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.cache;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.enterprise.util.Nonbinding;
+import javax.inject.Qualifier;
+
+/**
+ * <p>
+ * Use this annotation to inject a {@link Cache} and interact with it programmatically.
+ * </p>
+ * <p>
+ * Code example:
+ * 
+ * <pre>
+ * {@literal @}ApplicationScoped
+ * public class CachedService {
+ * 
+ *     {@literal @}CacheName("my-cache")
+ *     Cache cache;
+ *     
+ *     // Interact with the cache.
+ * }
+ * </pre>
+ * </p>
+ * 
+ * @see CacheManager
+ */
+@Qualifier
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ FIELD, METHOD, PARAMETER })
```

Comment:
Good catch! It is allowed on methods because of `CacheProducer` but I should add a check in `CacheProcessor` to forbid the annotation on any other method than the one from the producer.

Reply:
I added the check. `@CacheName` is now only allowed on methods from `io.quarkus.cache.runtime.CacheProducer`.

Reply:
On second thoughts, I think @CacheName should allow injections when placed on a method (like @Inject). I'll change that later... :)